#!/usr/bin/env python
PACKAGE = "target_tracking_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("kp",    double_t,    0, "Proportional", .0,  0, 10)
gen.add("ki",    double_t,    0, "Integral",    .0,   0, 10)
gen.add("deriv_prediction_dt",    double_t,    0, "Derivative prediction time interval",  0.1,   0, 1)
gen.add("max_deriv_noise_gain",    double_t,    0, "Max derivative noise gain",  1.0,   0, 10)

gen.add("manual_deriv_filter_weights",  bool_t,    0, "Manually tune the weights of the deriv filter",  False)
gen.add("alpha",    double_t,    0, "Weight on the previous derivative estimate in filtering",  0.5,   0, 1)

exit(gen.generate(PACKAGE, "target_tracking_controller", "PIDControllerParams"))
